src/v/pandaproxy/test/kafka_client_cache.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "pandaproxy/kafka_client_cache.h"
+
+#include "config/node_config.h"
+#include "config/rest_authn_endpoint.h"
+#include "kafka/client/configuration.h"
+#include "pandaproxy/client_cache_error.h"
+#include "pandaproxy/types.h"
+
+#include <seastar/core/sleep.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/unit_test.hpp>
+
+#include <chrono>
+#include <vector>
+
+using namespace std::chrono_literals;
+
+namespace pandaproxy {
+class test_client_cache : public kafka_client_cache {
```

Comment:
This is the commit where we should include `friend class test_client_cache` on the kafka client cache.